### DISCUSSION TITLE
[QUESTION] How to use Metal on iOS?

### DISCUSSION THREAD

**Discussion by thisisthekap** [mono/SkiaSharp#2103] [Q&A]
Which components should we use (SKGLView, SKCanvasView, ...) to use metal on iOS?

**Comment by mattleibow** [comment]
This depends on what you are using:
 - SkiaSharp.Views (native) - use SKMetalView
 - SkiaSharp.Views.Forms - unfortunately I have not created those view yet. You may be able to copy the renderer code and swap out the native views.

For both Forms and Maui, I want to create a new SKHardwareView or SKGpuView and then make the backend switch out depending on platforms. For example Metal on Apple, GL on Android and DX on WIndows.

The API is basically the same, but does not use the `SKPaintGLSurfaceEventArgs` args type and does not have GL-specific code in it.

I created a new issue to track this specifically: #2104
